### PR TITLE
Release ink! 3.0.0-rc7

### PR DESCRIPTION
**Pull request by cmichi**
[Preview here](https://github.com/paritytech/ink/blob/cmichi-release-ink-rc7/RELEASES.md).

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,53 +1,72 @@
...
       definition and ink! trait message respectively.
+- We switched to Rust edition 2021 ‒ [#977](https://github.com/paritytech/ink/pull/977).
+- Update chain extension example to show argument passing ‒ [#1029](https://github.com/paritytech/ink/pull/1029).
 
 ## Fixed
-- Contracts that are compiled as root (the default) now properly revert the transaction if a message
-  returned `Result::Err`.
-    - This does not apply to ink! smart contracts that are used as dependencies. Therefore it is still possible to match against a result return type for a called dependency.
+- Contracts now revert the transaction if an ink! message returns `Result::Err` ‒ [#975](https://github.com/paritytech/ink/pull/975), [#998](https://github.com/paritytech/ink/pull/998).
+    - **Note:** This does not apply to ink! smart contracts which are used as dependencies of a contract!<br/>
```
Comment (Robbepop) [inline RELEASES.md:62]: This was a bug and has been fixed. No longer holds true.

**Review by Robbepop — APPROVED** [review]
LGTM

File: RELEASES.md (hunk 1 above)
Comment (cmichi) [reply RELEASES.md:62]: So if the `adder` in our `delegator` example hypothetically would return `Result::Err` the entire transaction is reverted? So no way to match on `Result` in the calling contract?

Comment (Robbepop) [reply RELEASES.md:65]: Not the entire transaction is reverted but the state transition of the Adder contract call.

**Comment by xgreenx** [comment]
I think that [bug](https://github.com/paritytech/ink/issues/982) is critical for that release because a workaround doesn't exist.

Maybe you have time to fix that or check https://github.com/paritytech/ink/pull/983 with my suggestion about the fix?

**Comment by Robbepop** [comment]
> I think that [bug](https://github.com/paritytech/ink/issues/982) is critical for that release because a workaround doesn't exist.
> 
> Maybe you have time to fix that or check https://github.com/paritytech/ink/pull/983 with my suggestion about the fix?

We can fix this bug for another RC. no need to delay ghis release imo.

**Comment by xgreenx** [comment]
> We can fix this bug for another RC. no need to delay ghis release imo.

For OpenBrush it means that we will use `rc6`=( If you agree with the suggested fix I will implement it today and you can release it=)

**Comment by Robbepop** [comment]
I will need more time for that. We will be trying to release another smaller release very soon. A smaller follow-up release makes sense anyways since this one is already quite huge.

**Comment by xgreenx** [comment]
Okey=( Hope the next RC will be soon=) Because that bug is a blocker for us

**Comment by Robbepop** [comment]
> Okey=( Hope the next RC will be soon=) Because that bug is a blocker for us

The issues will be put on top of my priority list. ;)

**Pull request merged**